rbac/management/role/relation_api_dual_write_handler.py

```diff
@@ -34,16 +38,116 @@ class DualWriteException(Exception):
     pass
 
 
+class ReplicationEventType(str, Enum):
+    """Replication event type."""
+
+    CREATE_SYSTEM_ROLE = "create_system_role"
+    UPDATE_SYSTEM_ROLE = "update_system_role"
+    DELETE_SYSTEM_ROLE = "delete_system_role"
+    CREATE_CUSTOM_ROLE = "create_custom_role"
+    UPDATE_CUSTOM_ROLE = "update_custom_role"
+    DELETE_CUSTOM_ROLE = "delete_custom_role"
+    ASSIGN_ROLE = "assign_role"
+    UNASSIGN_ROLE = "unassign_role"
+    CREATE_GROUP = "create_group"
+    UPDATE_GROUP = "update_group"
+    DELETE_GROUP = "delete_group"
+
+
+class ReplicationEvent:
+    """What tuples changes to replicate."""
+
+    type: ReplicationEventType
+    partition_key: str
+    add: list[common_pb2.Relationship]
+    remove: list[common_pb2.Relationship]
+
+    def __init__(
+        self,
+        type: ReplicationEventType,
+        partition_key: str,
+        add: list[common_pb2.Relationship] = [],
+        remove: list[common_pb2.Relationship] = [],
+    ):
+        """Initialize ReplicationEvent."""
+        self.partition_key = partition_key
+        self.type = type
+        self.add = add
+        self.remove = remove
+
+
+class RelationReplicator(ABC):
+    """Type responsible for replicating relations to Kessel Relations."""
+
+    @abstractmethod
+    def replicate(self, event: ReplicationEvent):
+        """Replicate the given event to Kessel Relations."""
+        pass
+
+
+class OutboxReplicater(RelationReplicator):
```

Comment:
typo ? Should it be OutboxReplicator  ? (as is in RelationReplicator)

Reply:
Nice catch! Thank you